src/gen-ts.ts

```diff
@@ -18,37 +19,55 @@ import {serializeTsDeclarations} from './ts-serialize';
 
 /**
  * Analyze all files in the given directory using Polymer Analyzer, and return
- * a TypeScript declarations document string that describes its contents.
+ * TypeScript declaration document strings in a map keyed by relative path.
  */
-export async function generateDeclarations(rootDir: string): Promise<string> {
+export async function generateDeclarations(rootDir: string):
+    Promise<Map<string, string>> {
   const a = new analyzer.Analyzer({
     urlLoader: new analyzer.FSUrlLoader(rootDir),
     urlResolver: new analyzer.PackageUrlResolver(),
   });
   const analysis = await a.analyzePackage();
-  const ast = analyzerToAst(analysis);
-  return serializeTsDeclarations(ast);
+  const outFiles = new Map<string, string>();
+  for (const tsDoc of analyzerToAst(analysis)) {
+    outFiles.set(tsDoc.path, serializeTsDeclarations(tsDoc));
+  }
+  return outFiles;
 }
 
 /**
- * Make a TypeScript declarations document from the given Polymer Analyzer
+ * Make TypeScript declaration documents from the given Polymer Analyzer
  * result.
  */
-function analyzerToAst(analysis: analyzer.Analysis): ts.Document {
-  const root: ts.Document = {
-    kind: 'document',
-    members: [],
-  };
-
-  for (const doc of analysis.getFeatures({kind: 'js-document'})) {
+function analyzerToAst(analysis: analyzer.Analysis): ts.Document[] {
+  const tsDocs = new Array<ts.Document>();
+  for (const analyzerDoc of analysis.getFeatures({kind: 'js-document'})) {
     // TODO This is a very crude exclusion rule.
-    if (doc.url.match(/(test|demo)\//)) {
+    if (analyzerDoc.url.match(/(test|demo)\//)) {
       continue;
     }
-    handleDocument(doc, root);
+    // TODO I think there can be many analyzer documents with the same URL,
```

Comment:
There can be at most one toplevel document at a URL, but yeah there can be multiple inline documents, and they should be merged here I think

Reply:
Done.